internal/attestation/crafter/materials/csaf.go

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright 2023 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package materials
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	api "github.com/chainloop-dev/chainloop/internal/attestation/crafter/api/attestation/v1"
+	"github.com/chainloop-dev/chainloop/internal/casclient"
+	"github.com/chainloop-dev/chainloop/internal/schemavalidators"
+	"github.com/rs/zerolog"
+)
+
+type CSAFCrafter struct {
+	backend *casclient.CASBackend
+	*crafterCommon
+}
+
+func NewCSAFCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend, l *zerolog.Logger) (*CSAFCrafter, error) {
```

Comment:
Is there any way we could differentiate these material types?

What I mean is that one of the purposes of a crafter is to make sure we offer guardrails to developers so they provide exactly what they were asked for.

So in this case, if SecOps asks for a `Security incident response` I should not accept a `vex` file for example.

Is there any way we can differentiate those profiles? I could think of 1) doing the generic validation and 2) check if they are the right kind of profile. Is what possible?

Reply:
I completely agree I believe it's a bit counterproductive right now. We could do several things:
- Generic validation: Current approach, we just get the material and if non of the desired ones, raise a generic error.
- Individual crafter specification that just call this one, each of them with the individual material check.
- Fail on the material check, try to infer the profile from the document. This is a best effort since the `document.category` is not a required field but rather a suggestion to be set upon documentation, example:
> The value of /document/category SHALL be csaf_security_incident_response.
> The value of /document/category SHALL be csaf_informational_advisory.
> The value of /document/category SHALL be csaf_security_advisory.
> The value of /document/category SHALL be csaf_vex.

Reply:
I've refactored it so each time uses a different `constructor` that validates the proper material type and fails with sense.